[BUG] -DUSE_TF_PYTHON_LIBS is useless when install C++ interface

### Bug summary

According to  #1744 , if -DUSE_TF_PYTHON_LIBS=TRUE, the libtensorflow_cc.so is unnecessary.

However, when I run ` cmake -DTENSORFLOW_ROOT=$tensorflow_root -DCMAKE_INSTALL_PREFIX=$deepmd_root -DUSE_TF_PYTHON_LIBS=TRUE -DUSE_CUDA_TOOLKIT=TRUE -DLAMMPS_SOURCE_ROOT=$lammps_plugin_source ..`
An error occurred as follows:

    root@406837b3ed61:/home/zhouyy/deepmd-kit/source/build# cmake -DTENSORFLOW_ROOT=$tensorflow_root -DCMAKE_INSTALL_PREFIX=$deepmd_root -DUSE_TF_PYTHON_LIBS=TRUE -DUSE_CUDA_TOOLKIT=TRUE -DLAMMPS_SOURCE_ROOT=$lammps_plugin_source ..
    -- The C compiler identification is GNU 9.4.0
    -- The CXX compiler identification is GNU 9.4.0
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Found Git: /usr/bin/git (found version "2.25.1") 
    -- Supported model version: 1.1 
    -- Found CUDA: /usr/local/cuda (found version "11.6") 
    -- Found CUDA in /usr/local/cuda, build nv GPU support
    -- Will not build AMD GPU support
    -- Found Python: /root/miniconda2/envs/dpa-1/bin/python3.1 (found version "3.10.4") found components: Interpreter Development 
    cp: '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow/include' and '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow/include' are the same file
    -- Enabled cpp interface build, looking for tensorflow_cc and tensorflow_framework
    CMake Error at cmake/Findtensorflow.cmake:100 (message):
      Not found lib/'tensorflow_cc' in
      '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow;/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow;/usr/;/usr/local/'
      You can manually set the tensorflow install path by -DTENSORFLOW_ROOT
    Call Stack (most recent call first):
      CMakeLists.txt:95 (find_package)
    
    
    -- Configuring incomplete, errors occurred!
    See also "/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeOutput.log".

Is this a bug? Or I misunderstood?

### DeePMD-kit Version

2.1.6.dev27+g00c1b9e

### TensorFlow Version

2.9.0

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

deepmd-kit/source/build/CMakeFiles/CMakeOutput.log:

        The system is: Linux - 3.10.0-1160.59.1.el7.x86_64 - x86_64
        Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
        Compiler: /usr/bin/cc 
        Build flags: 
        Id flags:  
        
        The output was:
        0
        
        
        Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"
        
        The C compiler identification is GNU, found in "/home/zhouyy/deepmd-kit/source/build/CMakeFiles/3.16.3/CompilerIdC/a.out"
        
        Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
        Compiler: /usr/bin/c++ 
        Build flags: 
        Id flags:  
        
        The output was:
        0
        
        
        Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "a.out"
        
        The CXX compiler identification is GNU, found in "/home/zhouyy/deepmd-kit/source/build/CMakeFiles/3.16.3/CompilerIdCXX/a.out"
        
        Determining if the C compiler works passed with the following output:
        Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp
        
        Run Build Command(s):/usr/bin/make cmTC_94ce9/fast && /usr/bin/make -f CMakeFiles/cmTC_94ce9.dir/build.make CMakeFiles/cmTC_94ce9.dir/build
        make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        Building C object CMakeFiles/cmTC_94ce9.dir/testCCompiler.c.o
        /usr/bin/cc    -o CMakeFiles/cmTC_94ce9.dir/testCCompiler.c.o   -c /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp/testCCompiler.c
        Linking C executable cmTC_94ce9
        /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_94ce9.dir/link.txt --verbose=1
        /usr/bin/cc      CMakeFiles/cmTC_94ce9.dir/testCCompiler.c.o  -o cmTC_94ce9 
        make[1]: Leaving directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        
        
        
        Detecting C compiler ABI info compiled with the following output:
        Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp
        
        Run Build Command(s):/usr/bin/make cmTC_71360/fast && /usr/bin/make -f CMakeFiles/cmTC_71360.dir/build.make CMakeFiles/cmTC_71360.dir/build
        make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        Building C object CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o
        /usr/bin/cc   -v -o CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o   -c /usr/share/cmake-3.16/Modules/CMakeCCompilerABI.c
        Using built-in specs.
        COLLECT_GCC=/usr/bin/cc
        OFFLOAD_TARGET_NAMES=nvptx-none:hsa
        OFFLOAD_TARGET_DEFAULT=1
        Target: x86_64-linux-gnu
        Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
        Thread model: posix
        gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64'
         /usr/lib/gcc/x86_64-linux-gnu/9/cc1 -quiet -v -imultiarch x86_64-linux-gnu /usr/share/cmake-3.16/Modules/CMakeCCompilerABI.c -quiet -dumpbase CMakeCCompilerABI.c -mtune=generic -march=x86-64 -auxbase-strip CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o -version -fasynchronous-unwind-tables -fstack-protector-strong -Wformat -Wformat-security -fstack-clash-protection -fcf-protection -o /tmp/ccXxUIqF.s
        GNU C17 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)
	        compiled by GNU C version 9.4.0, GMP version 6.2.0, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.22.1-GMP
        
        GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
        ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
        ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/include-fixed"
        ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include"
        #include "..." search starts here:
        #include <...> search starts here:
         /usr/lib/gcc/x86_64-linux-gnu/9/include
         /usr/local/include
         /usr/include/x86_64-linux-gnu
         /usr/include
        End of search list.
        GNU C17 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)
	        compiled by GNU C version 9.4.0, GMP version 6.2.0, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.22.1-GMP
        
        GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
        Compiler executable checksum: c0c95c0b4209efec1c1892d5ff24030b
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64'
         as -v --64 -o CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o /tmp/ccXxUIqF.s
        GNU assembler version 2.34 (x86_64-linux-gnu) using BFD version (GNU Binutils for Ubuntu) 2.34
        COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
        LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64'
        Linking C executable cmTC_71360
        /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_71360.dir/link.txt --verbose=1
        /usr/bin/cc     -v CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o  -o cmTC_71360 
        Using built-in specs.
        COLLECT_GCC=/usr/bin/cc
        COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
        OFFLOAD_TARGET_NAMES=nvptx-none:hsa
        OFFLOAD_TARGET_DEFAULT=1
        Target: x86_64-linux-gnu
        Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
        Thread model: posix
        gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
        COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
        LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
        COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_71360' '-mtune=generic' '-march=x86-64'
         /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccoz3vV9.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o cmTC_71360 /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/local/cuda/lib64/stubs -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o
        COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_71360' '-mtune=generic' '-march=x86-64'
        make[1]: Leaving directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        
        
        
        Parsed C implicit include dir info from above output: rv=done
          found start of include info
          found start of implicit include info
            add: [/usr/lib/gcc/x86_64-linux-gnu/9/include]
            add: [/usr/local/include]
            add: [/usr/include/x86_64-linux-gnu]
            add: [/usr/include]
          end of search list found
          collapse include dir [/usr/lib/gcc/x86_64-linux-gnu/9/include] ==> [/usr/lib/gcc/x86_64-linux-gnu/9/include]
          collapse include dir [/usr/local/include] ==> [/usr/local/include]
          collapse include dir [/usr/include/x86_64-linux-gnu] ==> [/usr/include/x86_64-linux-gnu]
          collapse include dir [/usr/include] ==> [/usr/include]
          implicit include dirs: [/usr/lib/gcc/x86_64-linux-gnu/9/include;/usr/local/include;/usr/include/x86_64-linux-gnu;/usr/include]
        
        
        Parsed C implicit link information from above output:
          link line regex: [^( *|.*[/\])(ld|CMAKE_LINK_STARTFILE-NOTFOUND|([^/\]+-)?ld|collect2)[^/\]*( |$)]
          ignore line: [Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp]
          ignore line: []
          ignore line: [Run Build Command(s):/usr/bin/make cmTC_71360/fast && /usr/bin/make -f CMakeFiles/cmTC_71360.dir/build.make CMakeFiles/cmTC_71360.dir/build]
          ignore line: [make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp']
          ignore line: [Building C object CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o]
          ignore line: [/usr/bin/cc   -v -o CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o   -c /usr/share/cmake-3.16/Modules/CMakeCCompilerABI.c]
          ignore line: [Using built-in specs.]
          ignore line: [COLLECT_GCC=/usr/bin/cc]
          ignore line: [OFFLOAD_TARGET_NAMES=nvptx-none:hsa]
          ignore line: [OFFLOAD_TARGET_DEFAULT=1]
          ignore line: [Target: x86_64-linux-gnu]
          ignore line: [Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c ada c++ go brig d fortran objc obj-c++ gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32 m64 mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu]
          ignore line: [Thread model: posix]
          ignore line: [gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) ]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64']
          ignore line: [ /usr/lib/gcc/x86_64-linux-gnu/9/cc1 -quiet -v -imultiarch x86_64-linux-gnu /usr/share/cmake-3.16/Modules/CMakeCCompilerABI.c -quiet -dumpbase CMakeCCompilerABI.c -mtune=generic -march=x86-64 -auxbase-strip CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o -version -fasynchronous-unwind-tables -fstack-protector-strong -Wformat -Wformat-security -fstack-clash-protection -fcf-protection -o /tmp/ccXxUIqF.s]
          ignore line: [GNU C17 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)]
          ignore line: [	compiled by GNU C version 9.4.0  GMP version 6.2.0  MPFR version 4.0.2  MPC version 1.1.0  isl version isl-0.22.1-GMP]
          ignore line: []
          ignore line: [GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072]
          ignore line: [ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"]
          ignore line: [ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/include-fixed"]
          ignore line: [ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include"]
          ignore line: [#include "..." search starts here:]
          ignore line: [#include <...> search starts here:]
          ignore line: [ /usr/lib/gcc/x86_64-linux-gnu/9/include]
          ignore line: [ /usr/local/include]
          ignore line: [ /usr/include/x86_64-linux-gnu]
          ignore line: [ /usr/include]
          ignore line: [End of search list.]
          ignore line: [GNU C17 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)]
          ignore line: [	compiled by GNU C version 9.4.0  GMP version 6.2.0  MPFR version 4.0.2  MPC version 1.1.0  isl version isl-0.22.1-GMP]
          ignore line: []
          ignore line: [GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072]
          ignore line: [Compiler executable checksum: c0c95c0b4209efec1c1892d5ff24030b]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64']
          ignore line: [ as -v --64 -o CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o /tmp/ccXxUIqF.s]
          ignore line: [GNU assembler version 2.34 (x86_64-linux-gnu) using BFD version (GNU Binutils for Ubuntu) 2.34]
          ignore line: [COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/]
          ignore line: [LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o' '-c' '-mtune=generic' '-march=x86-64']
          ignore line: [Linking C executable cmTC_71360]
          ignore line: [/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_71360.dir/link.txt --verbose=1]
          ignore line: [/usr/bin/cc     -v CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o  -o cmTC_71360 ]
          ignore line: [Using built-in specs.]
          ignore line: [COLLECT_GCC=/usr/bin/cc]
          ignore line: [COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper]
          ignore line: [OFFLOAD_TARGET_NAMES=nvptx-none:hsa]
          ignore line: [OFFLOAD_TARGET_DEFAULT=1]
          ignore line: [Target: x86_64-linux-gnu]
          ignore line: [Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c ada c++ go brig d fortran objc obj-c++ gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32 m64 mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu]
          ignore line: [Thread model: posix]
          ignore line: [gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) ]
          ignore line: [COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/]
          ignore line: [LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_71360' '-mtune=generic' '-march=x86-64']
          link line: [ /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccoz3vV9.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o cmTC_71360 /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/local/cuda/lib64/stubs -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o]
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/collect2] ==> ignore
            arg [-plugin] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so] ==> ignore
            arg [-plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper] ==> ignore
            arg [-plugin-opt=-fresolution=/tmp/ccoz3vV9.res] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
            arg [-plugin-opt=-pass-through=-lc] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
            arg [--build-id] ==> ignore
            arg [--eh-frame-hdr] ==> ignore
            arg [-m] ==> ignore
            arg [elf_x86_64] ==> ignore
            arg [--hash-style=gnu] ==> ignore
            arg [--as-needed] ==> ignore
            arg [-dynamic-linker] ==> ignore
            arg [/lib64/ld-linux-x86-64.so.2] ==> ignore
            arg [-pie] ==> ignore
            arg [-znow] ==> ignore
            arg [-zrelro] ==> ignore
            arg [-o] ==> ignore
            arg [cmTC_71360] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o] ==> ignore
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib]
            arg [-L/lib/x86_64-linux-gnu] ==> dir [/lib/x86_64-linux-gnu]
            arg [-L/lib/../lib] ==> dir [/lib/../lib]
            arg [-L/usr/lib/x86_64-linux-gnu] ==> dir [/usr/lib/x86_64-linux-gnu]
            arg [-L/usr/lib/../lib] ==> dir [/usr/lib/../lib]
            arg [-L/usr/local/cuda/lib64/stubs] ==> dir [/usr/local/cuda/lib64/stubs]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../..] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../..]
            arg [CMakeFiles/cmTC_71360.dir/CMakeCCompilerABI.c.o] ==> ignore
            arg [-lgcc] ==> lib [gcc]
            arg [--push-state] ==> ignore
            arg [--as-needed] ==> ignore
            arg [-lgcc_s] ==> lib [gcc_s]
            arg [--pop-state] ==> ignore
            arg [-lc] ==> lib [c]
            arg [-lgcc] ==> lib [gcc]
            arg [--push-state] ==> ignore
            arg [--as-needed] ==> ignore
            arg [-lgcc_s] ==> lib [gcc_s]
            arg [--pop-state] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o] ==> ignore
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9] ==> [/usr/lib/gcc/x86_64-linux-gnu/9]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu] ==> [/usr/lib/x86_64-linux-gnu]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib] ==> [/usr/lib]
          collapse library dir [/lib/x86_64-linux-gnu] ==> [/lib/x86_64-linux-gnu]
          collapse library dir [/lib/../lib] ==> [/lib]
          collapse library dir [/usr/lib/x86_64-linux-gnu] ==> [/usr/lib/x86_64-linux-gnu]
          collapse library dir [/usr/lib/../lib] ==> [/usr/lib]
          collapse library dir [/usr/local/cuda/lib64/stubs] ==> [/usr/local/cuda/lib64/stubs]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../..] ==> [/usr/lib]
          implicit libs: [gcc;gcc_s;c;gcc;gcc_s]
          implicit dirs: [/usr/lib/gcc/x86_64-linux-gnu/9;/usr/lib/x86_64-linux-gnu;/usr/lib;/lib/x86_64-linux-gnu;/lib;/usr/local/cuda/lib64/stubs]
          implicit fwks: []
        
        
        Determining if the CXX compiler works passed with the following output:
        Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp
        
        Run Build Command(s):/usr/bin/make cmTC_9980f/fast && /usr/bin/make -f CMakeFiles/cmTC_9980f.dir/build.make CMakeFiles/cmTC_9980f.dir/build
        make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        Building CXX object CMakeFiles/cmTC_9980f.dir/testCXXCompiler.cxx.o
        /usr/bin/c++     -o CMakeFiles/cmTC_9980f.dir/testCXXCompiler.cxx.o -c /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
        Linking CXX executable cmTC_9980f
        /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_9980f.dir/link.txt --verbose=1
        /usr/bin/c++       CMakeFiles/cmTC_9980f.dir/testCXXCompiler.cxx.o  -o cmTC_9980f 
        make[1]: Leaving directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        
        
        
        Detecting CXX compiler ABI info compiled with the following output:
        Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp
        
        Run Build Command(s):/usr/bin/make cmTC_fc72d/fast && /usr/bin/make -f CMakeFiles/cmTC_fc72d.dir/build.make CMakeFiles/cmTC_fc72d.dir/build
        make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        Building CXX object CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o
        /usr/bin/c++    -v -o CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -c /usr/share/cmake-3.16/Modules/CMakeCXXCompilerABI.cpp
        Using built-in specs.
        COLLECT_GCC=/usr/bin/c++
        OFFLOAD_TARGET_NAMES=nvptx-none:hsa
        OFFLOAD_TARGET_DEFAULT=1
        Target: x86_64-linux-gnu
        Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
        Thread model: posix
        gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
         /usr/lib/gcc/x86_64-linux-gnu/9/cc1plus -quiet -v -imultiarch x86_64-linux-gnu -D_GNU_SOURCE /usr/share/cmake-3.16/Modules/CMakeCXXCompilerABI.cpp -quiet -dumpbase CMakeCXXCompilerABI.cpp -mtune=generic -march=x86-64 -auxbase-strip CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -version -fasynchronous-unwind-tables -fstack-protector-strong -Wformat -Wformat-security -fstack-clash-protection -fcf-protection -o /tmp/ccpC6IRF.s
        GNU C++14 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)
	        compiled by GNU C version 9.4.0, GMP version 6.2.0, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.22.1-GMP
        
        GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
        ignoring duplicate directory "/usr/include/x86_64-linux-gnu/c++/9"
        ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
        ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/include-fixed"
        ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include"
        #include "..." search starts here:
        #include <...> search starts here:
         /usr/include/c++/9
         /usr/include/x86_64-linux-gnu/c++/9
         /usr/include/c++/9/backward
         /usr/lib/gcc/x86_64-linux-gnu/9/include
         /usr/local/include
         /usr/include/x86_64-linux-gnu
         /usr/include
        End of search list.
        GNU C++14 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)
	        compiled by GNU C version 9.4.0, GMP version 6.2.0, MPFR version 4.0.2, MPC version 1.1.0, isl version isl-0.22.1-GMP
        
        GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072
        Compiler executable checksum: 65fe925b83d3956b533de4aaba7dace0
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
         as -v --64 -o CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o /tmp/ccpC6IRF.s
        GNU assembler version 2.34 (x86_64-linux-gnu) using BFD version (GNU Binutils for Ubuntu) 2.34
        COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
        LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
        COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
        Linking CXX executable cmTC_fc72d
        /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_fc72d.dir/link.txt --verbose=1
        /usr/bin/c++      -v CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o  -o cmTC_fc72d 
        Using built-in specs.
        COLLECT_GCC=/usr/bin/c++
        COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
        OFFLOAD_TARGET_NAMES=nvptx-none:hsa
        OFFLOAD_TARGET_DEFAULT=1
        Target: x86_64-linux-gnu
        Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
        Thread model: posix
        gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) 
        COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/
        LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/
        COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_fc72d' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
         /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccmHMpa4.res -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o cmTC_fc72d /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/local/cuda/lib64/stubs -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o
        COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_fc72d' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
        make[1]: Leaving directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp'
        
        
        
        Parsed CXX implicit include dir info from above output: rv=done
          found start of include info
          found start of implicit include info
            add: [/usr/include/c++/9]
            add: [/usr/include/x86_64-linux-gnu/c++/9]
            add: [/usr/include/c++/9/backward]
            add: [/usr/lib/gcc/x86_64-linux-gnu/9/include]
            add: [/usr/local/include]
            add: [/usr/include/x86_64-linux-gnu]
            add: [/usr/include]
          end of search list found
          collapse include dir [/usr/include/c++/9] ==> [/usr/include/c++/9]
          collapse include dir [/usr/include/x86_64-linux-gnu/c++/9] ==> [/usr/include/x86_64-linux-gnu/c++/9]
          collapse include dir [/usr/include/c++/9/backward] ==> [/usr/include/c++/9/backward]
          collapse include dir [/usr/lib/gcc/x86_64-linux-gnu/9/include] ==> [/usr/lib/gcc/x86_64-linux-gnu/9/include]
          collapse include dir [/usr/local/include] ==> [/usr/local/include]
          collapse include dir [/usr/include/x86_64-linux-gnu] ==> [/usr/include/x86_64-linux-gnu]
          collapse include dir [/usr/include] ==> [/usr/include]
          implicit include dirs: [/usr/include/c++/9;/usr/include/x86_64-linux-gnu/c++/9;/usr/include/c++/9/backward;/usr/lib/gcc/x86_64-linux-gnu/9/include;/usr/local/include;/usr/include/x86_64-linux-gnu;/usr/include]
        
        
        Parsed CXX implicit link information from above output:
          link line regex: [^( *|.*[/\])(ld|CMAKE_LINK_STARTFILE-NOTFOUND|([^/\]+-)?ld|collect2)[^/\]*( |$)]
          ignore line: [Change Dir: /home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp]
          ignore line: []
          ignore line: [Run Build Command(s):/usr/bin/make cmTC_fc72d/fast && /usr/bin/make -f CMakeFiles/cmTC_fc72d.dir/build.make CMakeFiles/cmTC_fc72d.dir/build]
          ignore line: [make[1]: Entering directory '/home/zhouyy/deepmd-kit/source/build/CMakeFiles/CMakeTmp']
          ignore line: [Building CXX object CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o]
          ignore line: [/usr/bin/c++    -v -o CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -c /usr/share/cmake-3.16/Modules/CMakeCXXCompilerABI.cpp]
          ignore line: [Using built-in specs.]
          ignore line: [COLLECT_GCC=/usr/bin/c++]
          ignore line: [OFFLOAD_TARGET_NAMES=nvptx-none:hsa]
          ignore line: [OFFLOAD_TARGET_DEFAULT=1]
          ignore line: [Target: x86_64-linux-gnu]
          ignore line: [Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c ada c++ go brig d fortran objc obj-c++ gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32 m64 mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu]
          ignore line: [Thread model: posix]
          ignore line: [gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) ]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64']
          ignore line: [ /usr/lib/gcc/x86_64-linux-gnu/9/cc1plus -quiet -v -imultiarch x86_64-linux-gnu -D_GNU_SOURCE /usr/share/cmake-3.16/Modules/CMakeCXXCompilerABI.cpp -quiet -dumpbase CMakeCXXCompilerABI.cpp -mtune=generic -march=x86-64 -auxbase-strip CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -version -fasynchronous-unwind-tables -fstack-protector-strong -Wformat -Wformat-security -fstack-clash-protection -fcf-protection -o /tmp/ccpC6IRF.s]
          ignore line: [GNU C++14 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)]
          ignore line: [	compiled by GNU C version 9.4.0  GMP version 6.2.0  MPFR version 4.0.2  MPC version 1.1.0  isl version isl-0.22.1-GMP]
          ignore line: []
          ignore line: [GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072]
          ignore line: [ignoring duplicate directory "/usr/include/x86_64-linux-gnu/c++/9"]
          ignore line: [ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"]
          ignore line: [ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/include-fixed"]
          ignore line: [ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include"]
          ignore line: [#include "..." search starts here:]
          ignore line: [#include <...> search starts here:]
          ignore line: [ /usr/include/c++/9]
          ignore line: [ /usr/include/x86_64-linux-gnu/c++/9]
          ignore line: [ /usr/include/c++/9/backward]
          ignore line: [ /usr/lib/gcc/x86_64-linux-gnu/9/include]
          ignore line: [ /usr/local/include]
          ignore line: [ /usr/include/x86_64-linux-gnu]
          ignore line: [ /usr/include]
          ignore line: [End of search list.]
          ignore line: [GNU C++14 (Ubuntu 9.4.0-1ubuntu1~20.04.1) version 9.4.0 (x86_64-linux-gnu)]
          ignore line: [	compiled by GNU C version 9.4.0  GMP version 6.2.0  MPFR version 4.0.2  MPC version 1.1.0  isl version isl-0.22.1-GMP]
          ignore line: []
          ignore line: [GGC heuristics: --param ggc-min-expand=100 --param ggc-min-heapsize=131072]
          ignore line: [Compiler executable checksum: 65fe925b83d3956b533de4aaba7dace0]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64']
          ignore line: [ as -v --64 -o CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o /tmp/ccpC6IRF.s]
          ignore line: [GNU assembler version 2.34 (x86_64-linux-gnu) using BFD version (GNU Binutils for Ubuntu) 2.34]
          ignore line: [COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/]
          ignore line: [LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o' '-c' '-shared-libgcc' '-mtune=generic' '-march=x86-64']
          ignore line: [Linking CXX executable cmTC_fc72d]
          ignore line: [/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_fc72d.dir/link.txt --verbose=1]
          ignore line: [/usr/bin/c++      -v CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o  -o cmTC_fc72d ]
          ignore line: [Using built-in specs.]
          ignore line: [COLLECT_GCC=/usr/bin/c++]
          ignore line: [COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper]
          ignore line: [OFFLOAD_TARGET_NAMES=nvptx-none:hsa]
          ignore line: [OFFLOAD_TARGET_DEFAULT=1]
          ignore line: [Target: x86_64-linux-gnu]
          ignore line: [Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c ada c++ go brig d fortran objc obj-c++ gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32 m64 mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu]
          ignore line: [Thread model: posix]
          ignore line: [gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1) ]
          ignore line: [COMPILER_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/]
          ignore line: [LIBRARY_PATH=/usr/lib/gcc/x86_64-linux-gnu/9/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib/:/lib/x86_64-linux-gnu/:/lib/../lib/:/usr/lib/x86_64-linux-gnu/:/usr/lib/../lib/:/usr/local/cuda/lib64/stubs/:/usr/lib/gcc/x86_64-linux-gnu/9/../../../:/lib/:/usr/lib/]
          ignore line: [COLLECT_GCC_OPTIONS='-v' '-o' 'cmTC_fc72d' '-shared-libgcc' '-mtune=generic' '-march=x86-64']
          link line: [ /usr/lib/gcc/x86_64-linux-gnu/9/collect2 -plugin /usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper -plugin-opt=-fresolution=/tmp/ccmHMpa4.res -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o cmTC_fc72d /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/local/cuda/lib64/stubs -L/usr/lib/gcc/x86_64-linux-gnu/9/../../.. CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o]
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/collect2] ==> ignore
            arg [-plugin] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/liblto_plugin.so] ==> ignore
            arg [-plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper] ==> ignore
            arg [-plugin-opt=-fresolution=/tmp/ccmHMpa4.res] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
            arg [-plugin-opt=-pass-through=-lc] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc_s] ==> ignore
            arg [-plugin-opt=-pass-through=-lgcc] ==> ignore
            arg [--build-id] ==> ignore
            arg [--eh-frame-hdr] ==> ignore
            arg [-m] ==> ignore
            arg [elf_x86_64] ==> ignore
            arg [--hash-style=gnu] ==> ignore
            arg [--as-needed] ==> ignore
            arg [-dynamic-linker] ==> ignore
            arg [/lib64/ld-linux-x86-64.so.2] ==> ignore
            arg [-pie] ==> ignore
            arg [-znow] ==> ignore
            arg [-zrelro] ==> ignore
            arg [-o] ==> ignore
            arg [cmTC_fc72d] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/Scrt1.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crti.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/crtbeginS.o] ==> ignore
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib]
            arg [-L/lib/x86_64-linux-gnu] ==> dir [/lib/x86_64-linux-gnu]
            arg [-L/lib/../lib] ==> dir [/lib/../lib]
            arg [-L/usr/lib/x86_64-linux-gnu] ==> dir [/usr/lib/x86_64-linux-gnu]
            arg [-L/usr/lib/../lib] ==> dir [/usr/lib/../lib]
            arg [-L/usr/local/cuda/lib64/stubs] ==> dir [/usr/local/cuda/lib64/stubs]
            arg [-L/usr/lib/gcc/x86_64-linux-gnu/9/../../..] ==> dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../..]
            arg [CMakeFiles/cmTC_fc72d.dir/CMakeCXXCompilerABI.cpp.o] ==> ignore
            arg [-lstdc++] ==> lib [stdc++]
            arg [-lm] ==> lib [m]
            arg [-lgcc_s] ==> lib [gcc_s]
            arg [-lgcc] ==> lib [gcc]
            arg [-lc] ==> lib [c]
            arg [-lgcc_s] ==> lib [gcc_s]
            arg [-lgcc] ==> lib [gcc]
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/crtendS.o] ==> ignore
            arg [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crtn.o] ==> ignore
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9] ==> [/usr/lib/gcc/x86_64-linux-gnu/9]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu] ==> [/usr/lib/x86_64-linux-gnu]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib] ==> [/usr/lib]
          collapse library dir [/lib/x86_64-linux-gnu] ==> [/lib/x86_64-linux-gnu]
          collapse library dir [/lib/../lib] ==> [/lib]
          collapse library dir [/usr/lib/x86_64-linux-gnu] ==> [/usr/lib/x86_64-linux-gnu]
          collapse library dir [/usr/lib/../lib] ==> [/usr/lib]
          collapse library dir [/usr/local/cuda/lib64/stubs] ==> [/usr/local/cuda/lib64/stubs]
          collapse library dir [/usr/lib/gcc/x86_64-linux-gnu/9/../../..] ==> [/usr/lib]
          implicit libs: [stdc++;m;gcc_s;gcc;c;gcc_s;gcc]
          implicit dirs: [/usr/lib/gcc/x86_64-linux-gnu/9;/usr/lib/x86_64-linux-gnu;/usr/lib;/lib/x86_64-linux-gnu;/lib;/usr/local/cuda/lib64/stubs]
          implicit fwks: []

### Steps to Reproduce

The following is my whole operations:

    conda create -n XXX python=3.10.4
    pip install tensorflow==2.9.0
    cd deepmd-kit-source-dir
    git checkout devel
    deepmd_source_dir=PWD
    export DP_VARIANT=cuda
    pip install . -i https://pypi.tuna.tsinghua.edu.cn/simple/
    cd source
    mkdir build
    cd build
    cmake -DTENSORFLOW_ROOT=$tensorflow_root -DCMAKE_INSTALL_PREFIX=$deepmd_root -DUSE_TF_PYTHON_LIBS=TRUE -DUSE_CUDA_TOOLKIT=TRUE -DLAMMPS_SOURCE_ROOT=$lammps_plugin_source ..

### Further Information, Files, and Links

_No response_

what's the directory of `tensorflow_root`?
```
-- Found Python: /root/miniconda2/envs/dpa-1/bin/python3.1 (found version "3.10.4") found components: Interpreter Development 
cp: '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow/include' and '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow/include' are the same file
-- Enabled cpp interface build, looking for tensorflow_cc and tensorflow_framework
CMake Error at cmake/Findtensorflow.cmake:100 (message):
  Not found lib/'tensorflow_cc' in
  '/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow;/root/miniconda2/envs/dpa-1/lib/python3.10/site-packages/tensorflow;/usr/;/usr/local/'
  You can manually set the tensorflow install path by -DTENSORFLOW_ROOT
Call Stack (most recent call first):
```
According to the above info, it seems like you have set the directory of tensorflow_root the same as the directory of python tensorflow installation path. Please consider changing the tensorflow_root path.

> 

> > Hi, I just wanted to add that I am hoping this bug gets fixed, because I am having issues compiling libtensorflow_cc.so at the cluster where I run DeepMD, and being able to use this flag would be greatly appreciated.
> > Just to note, I install tensorflow using conda from OpenCE repository (https://osuosl.org/services/powerdev/opence/). The cluster I run at is **linux-ppc64le** architecture.
> 
> Are you facing the same error as this issue? Please provide more specific information about the task error, including how to reproduce your problem.

Well in the previous installations (i tried many different combinations) I think I was getting the same error, but now when I run the following installation sequence of commands:

####Compiling DeepMD on RPI DCS

#download lammps
cd /gpfs/u/scratch/THML/THMLkrrm && git clone https://github.com/lammps/lammps.git

#install deepmd
git clone https://github.com/deepmodeling/deepmd-kit.git && cd deepmd-kit && deepmd_source_dir=`pwd` && conda create --name opence_deepmd python=3.8 -y && conda activate opence_deepmd && conda install -c https://ftp.osuosl.org/pub/open-ce/current/ tensorflow horovod cudatoolkit-dev-y && export DP_VARIANT=cuda && export CUDA_TOOLKIT_ROOT_DIR=/gpfs/u/scratch/THML/shared/miniconda3/envs/opence_deepmd/pkgs/cuda-toolkit && pip install . && dp -h && horovodrun --check-build && cd $deepmd_source_dir/source && mkdir build && cd build/ && cmake -DTENSORFLOW_ROOT=/gpfs/u/scratch/THML/shared/miniconda3/envs/opence_deepmd/lib/python3.8/site-packages/tensorflow -DUSE_TF_PYTHON_LIBS=TRUE -DCMAKE_INSTALL_PREFIX=/gpfs/u/scratch/THML/THMLkrrm/deepmd-kit/install_folder -DLAMMPS_SOURCE_ROOT=/gpfs/u/scratch/THML/THMLkrrm/lammps/ .. && make -j16 install && cd ../../install_folder && deepmd_root=`pwd`

I get the following error:

-- The C compiler identification is GNU 8.4.1
-- The CXX compiler identification is GNU 8.4.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.27.0") 
-- Supported model version: 1.1 
-- Will not build nv GPU support
-- Will not build AMD GPU support
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Python (missing: Python_INCLUDE_DIRS Python_LIBRARIES
  Development Development.Module Development.Embed) (found version "2.7.18")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindPython.cmake:436 (find_package_handle_standard_args)
  CMakeLists.txt:84 (find_package)


-- Configuring incomplete, errors occurred!
See also "/gpfs/u/home/THML/THMLkrrm/scratch/deepmd-kit/source/build/CMakeFiles/CMakeOutput.log".

> Change the `$TENSORFLOW_ROOT` to another empty folder. The current implementation copies the TF include files as well as the libraries from the python site-packages to your specified `$TENSORFLOW_ROOT` directory. So don't make them to the same.

I just did that and got the following error (but now it is saying it cant find tensorflow files, gpfs/u/home/THML/THMLkrrm/scratch/tensorflow_deepmd is supposed to be the empty folder):

-- The C compiler identification is GNU 8.4.1
-- The CXX compiler identification is GNU 8.4.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.27.0") 
-- Supported model version: 1.1 
-- Will not build nv GPU support
-- Will not build AMD GPU support
-- Found Python: /gpfs/u/scratch/THML/shared/miniconda3/bin/python3.9 (found version "3.9.12") found components: Interpreter Development Development.Module Development.Embed 
/gpfs/u/home/THML/THMLkrrm/scratch/deepmd-kit/source/install: /gpfs/u/scratch/THML/shared/miniconda3/lib/python3.9/site-packages/tensorflow, TensorFlow not found!
CMake Error at cmake/Findtensorflow.cmake:84 (message):
  Not found 'include/tensorflow/core/public/session.h' directory or other
  header files in path
  '/gpfs/u/home/THML/THMLkrrm/scratch/tensorflow_deepmd;/gpfs/u/home/THML/THMLkrrm/scratch/tensorflow_deepmd;/usr/;/usr/local/'
  You can manually set the tensorflow install path by -DTENSORFLOW_ROOT
Call Stack (most recent call first):
  CMakeLists.txt:88 (find_package)


-- Configuring incomplete, errors occurred!
See also "/gpfs/u/home/THML/THMLkrrm/scratch/deepmd-kit/source/build/CMakeFiles/CMakeOutput.log".


Am I maybe misunderstanding something here?

Well, I tried to reproduce your installation progress, and got a different error(by using the latest devel branch):
```
cmake -DUSE_CUDA_TOOLKIT=true -DTENSORFLOW_ROOT=/home/ldh/deepmd-kit/tensorflow_root -DCMAKE_INSTALL_PREFIX=/home/ldh/deepmd-kit/deepmd_root -DUSE_TF_PYTHON_LIBS=true ..
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.25.1")
-- Supported model version: 1.1
-- Found CUDA: /home/ldh/anaconda3/envs/opence_deepmd (found version "11.4")
-- Found CUDA in /home/ldh/anaconda3/envs/opence_deepmd, build nv GPU support
-- Will not build AMD GPU support
-- Found Python: /home/ldh/anaconda3/envs/opence_deepmd/bin/python3.9 (found version "3.9.15") found components: Interpreter Development Development.Module Development.Embed
-- Enabled cpp interface build, looking for tensorflow_cc and tensorflow_framework
CMake Warning (dev) at cmake/Findtensorflow.cmake:161 (file):
  You have used file(GET_RUNTIME_DEPENDENCIES) in project mode.  This is
  probably not what you intended to do.  Instead, please consider using it in
  an install(CODE) or install(SCRIPT) command.  For example:

    install(CODE [[
      file(GET_RUNTIME_DEPENDENCIES
        # ...
        )
      ]])
Call Stack (most recent call first):
  CMakeLists.txt:119 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Protobuf headers are not found in the directory of TensorFlow, assuming external protobuf was used to build TensorFlow
CMake Error at cmake/Findtensorflow.cmake:179 (message):
  TensorFlow is not linked to protobuf
Call Stack (most recent call first):
  CMakeLists.txt:119 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/ldh/deepmd-kit/conda-dp/deepmd-kit/source/build/CMakeFiles/CMakeOutput.log".
```
Have you tried to use virtualenv to install tensorflow and compile deepmd-kit?

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -369,6 +369,6 @@ if(BUILD_CPP_IF)
                             -D_GLIBCXX_USE_CXX11_ABI=${OP_CXX_ABI})
   if (USE_TF_PYTHON_LIBS)
     # link: libpython3.x.so
-    target_link_libraries (TensorFlow::tensorflow_cc INTERFACE Python::python)
+    target_link_libraries (TensorFlow::tensorflow_cc INTERFACE ${Python_LIBRARIES})
   endif()
 endif()

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -139,7 +139,7 @@ foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
   if (TensorFlowFramework_LIBRARY_${module})
     list(APPEND TensorFlowFramework_LIBRARY ${TensorFlowFramework_LIBRARY_${module}})
     get_filename_component(TensorFlowFramework_LIBRARY_PATH_${module} ${TensorFlowFramework_LIBRARY_${module}} PATH)
-    list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
+    list(APPEND TensorFlowFramework_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
   elseif (tensorflow_FIND_REQUIRED)
     message(FATAL_ERROR 
       "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' "

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -32,16 +32,18 @@ if (BUILD_CPP_IF AND INSTALL_TENSORFLOW)
 	  )
 endif ()
 
-if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
+if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS AND NOT SKBUILD)
   # Here we try to install libtensorflow_cc.so as well as libtensorflow_framework.so using libs within the python site-package tensorflow folder.
-
-  if (NOT DEFINED TENSORFLOW_ROOT)
-    set (TENSORFLOW_ROOT ${CMAKE_INSTALL_PREFIX})
-  endif ()
-  # execute install script
   execute_process(
-    COMMAND sh ${DEEPMD_SOURCE_DIR}/source/install/install_tf.sh ${Python_SITELIB} ${TENSORFLOW_ROOT}
-    )
+    COMMAND "${PYTHON_EXECUTABLE}" -c "import tensorflow; print(tensorflow.sysconfig.get_lib())"
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    OUTPUT_VARIABLE TENSORFLOW_ROOT
+    RESULT_VARIABLE TENSORFLOW_ROOT_RESULT_VAR
+    ERROR_VARIABLE TENSORFLOW_ROOT_ERROR_VAR
+  )
+  if (NOT ${TENSORFLOW_ROOT_RESULT_VAR} EQUAL 0)
+    message(FATAL_ERROR "Cannot determine tensorflow root, error message: ${TENSORFLOW_ROOT_ERROR_VAR}")
+  endif()
 endif ()
 
 if(DEFINED TENSORFLOW_ROOT)
@@ -50,12 +52,12 @@ endif(DEFINED TENSORFLOW_ROOT)
 
 # define the search path
 list(APPEND TensorFlow_search_PATHS ${TENSORFLOW_ROOT})
-if(BUILD_CPP_IF)
+if(BUILD_CPP_IF AND NOT USE_TF_PYTHON_LIBS)
 list(APPEND TensorFlow_search_PATHS ${TENSORFLOW_ROOT_NO64})
 list(APPEND TensorFlow_search_PATHS "/usr/")
 list(APPEND TensorFlow_search_PATHS "/usr/local/")
 endif()
-if(BUILD_PY_IF)
+if(BUILD_PY_IF OR USE_TF_PYTHON_LIBS)
 	# here TENSORFLOW_ROOT is path to site-packages/tensorflow
 	# for conda libraries, append extra paths
 	list(APPEND TensorFlow_search_PATHS "${TENSORFLOW_ROOT}/../tensorflow_core")
@@ -80,7 +82,7 @@ if (NOT TensorFlow_INCLUDE_DIRS AND tensorflow_FIND_REQUIRED)
     "You can manually set the tensorflow install path by -DTENSORFLOW_ROOT ")
 endif ()
 
-if (BUILD_CPP_IF)
+if (BUILD_CPP_IF AND NOT USE_TF_PYTHON_LIBS)
   message (STATUS "Enabled cpp interface build, looking for tensorflow_cc and tensorflow_framework")
   # tensorflow_cc and tensorflow_framework
   if (NOT TensorFlow_FIND_COMPONENTS)
@@ -127,22 +129,43 @@ else ()
 list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .so.1)
 list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .so.2)
 endif()
-set (TensorFlowFramework_LIBRARY_PATH "")
+set (TensorFlow_LIBRARY_PATH "")
 foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
-  find_library(TensorFlowFramework_LIBRARY_${module}
+  find_library(TensorFlow_LIBRARY_${module}
     NAMES ${module}
     PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
     )
-  if (TensorFlowFramework_LIBRARY_${module})
-    list(APPEND TensorFlowFramework_LIBRARY ${TensorFlowFramework_LIBRARY_${module}})
-    get_filename_component(TensorFlowFramework_LIBRARY_PATH_${module} ${TensorFlowFramework_LIBRARY_${module}} PATH)
-    list(APPEND TensorFlowFramework_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
+  if (TensorFlow_LIBRARY_${module})
+    list(APPEND TensorFlow_LIBRARY ${TensorFlow_LIBRARY_${module}})
+    get_filename_component(TensorFlow_LIBRARY_PATH_${module} ${TensorFlow_LIBRARY_${module}} PATH)
+    list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlow_LIBRARY_PATH_${module}})
   elseif (tensorflow_FIND_REQUIRED)
     message(FATAL_ERROR 
-      "Not found lib/'${module}' in '${TensorFlow_search_PATHS}' "
+      "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' "
       "You can manually set the tensorflow install path by -DTENSORFLOW_ROOT ")
   endif ()
 endforeach ()
+
+if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
+  list(APPEND TensorFlow_search_PATHS ${TENSORFLOW_ROOT}/python)
+  set(TF_SUFFIX "")
+  set(TensorFlowFramework_FIND_COMPONENTS _pywrap_tensorflow_internal)
+  foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
+    find_library(TensorFlowFramework_LIBRARY_${module}
+      NAMES ${module}
+      PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
+      )
+    if (TensorFlowFramework_LIBRARY_${module})
+      list(APPEND TensorFlowFramework_LIBRARY ${TensorFlowFramework_LIBRARY_${module}})
+      get_filename_component(TensorFlowFramework_LIBRARY_PATH_${module} ${TensorFlowFramework_LIBRARY_${module}} PATH)
+      list(APPEND TensorFlowFramework_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
+    elseif (tensorflow_FIND_REQUIRED)
+      message(FATAL_ERROR 
+        "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' ")
+    endif ()
+  endforeach ()
+endif()
+
 
 # find protobuf header
 find_path(TensorFlow_INCLUDE_DIRS_GOOGLE
@@ -346,6 +369,6 @@ if(BUILD_CPP_IF)
                             -D_GLIBCXX_USE_CXX11_ABI=${OP_CXX_ABI})
   if (USE_TF_PYTHON_LIBS)
     # link: libpython3.x.so
-    target_link_libraries (TensorFlow::tensorflow_cc INTERFACE ${Python_LIBRARIES})
+    target_link_libraries (TensorFlow::tensorflow_cc INTERFACE Python::python)
   endif()
 endif()

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -147,6 +147,7 @@ foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
   endif ()
 endforeach ()
 
+# find _pywrap_tensorflow_internal and set it as tensorflow_cc
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
   set(TF_SUFFIX python)
   set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal.so)
@@ -165,6 +166,7 @@ if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
         "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' ")
     endif ()
   endforeach ()
+  set (TensorFlow_LIBRARY_tensorflow_cc ${TensorFlow_LIBRARY__pywrap_tensorflow_internal})
 endif()
 
 

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -142,27 +142,27 @@ foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
     list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
   elseif (tensorflow_FIND_REQUIRED)
     message(FATAL_ERROR 
-      "Not found ${TF_SUFFIX}${module} in '${TensorFlow_search_PATHS}' "
+      "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' "
       "You can manually set the tensorflow install path by -DTENSORFLOW_ROOT ")
   endif ()
 endforeach ()
 
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
-  list(APPEND TensorFlow_search_PATHS ${TENSORFLOW_ROOT}/python)
-  set(TF_SUFFIX "")
-  set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal)
+  set(TF_SUFFIX python)
+  set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal.so)
   foreach (module ${TensorFlow_FIND_COMPONENTS})
     find_library(TensorFlow_LIBRARY_${module}
       NAMES ${module}
       PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
       )
+    message(STATUS ${TensorFlow_LIBRARY_${module}})
     if (TensorFlow_LIBRARY_${module})
       list(APPEND TensorFlow_LIBRARY ${TensorFlow_LIBRARY_${module}})
       get_filename_component(TensorFlow_LIBRARY_PATH_${module} ${TensorFlow_LIBRARY_${module}} PATH)
       list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlow_LIBRARY_PATH_${module}})
     elseif (tensorflow_FIND_REQUIRED)
       message(FATAL_ERROR 
-        "Not found ${TF_SUFFIX}${module} in '${TensorFlow_search_PATHS}' ")
+        "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' ")
     endif ()
   endforeach ()
 endif()

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -150,7 +150,7 @@ endforeach ()
 # find _pywrap_tensorflow_internal and set it as tensorflow_cc
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
   set(TF_SUFFIX python)
-  set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal.so)
+  set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal${CMAKE_SHARED_LIBRARY_SUFFIX})
   foreach (module ${TensorFlow_FIND_COMPONENTS})
     find_library(TensorFlow_LIBRARY_${module}
       NAMES ${module}

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -35,11 +35,12 @@ endif ()
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS AND NOT SKBUILD)
   # Here we try to install libtensorflow_cc.so as well as libtensorflow_framework.so using libs within the python site-package tensorflow folder.
   execute_process(
-    COMMAND "${PYTHON_EXECUTABLE}" -c "import tensorflow; print(tensorflow.sysconfig.get_lib())"
+    COMMAND ${PYTHON_EXECUTABLE} -c "import tensorflow; print(tensorflow.sysconfig.get_lib())"
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
     OUTPUT_VARIABLE TENSORFLOW_ROOT
     RESULT_VARIABLE TENSORFLOW_ROOT_RESULT_VAR
     ERROR_VARIABLE TENSORFLOW_ROOT_ERROR_VAR
+    OUTPUT_STRIP_TRAILING_WHITESPACE
   )
   if (NOT ${TENSORFLOW_ROOT_RESULT_VAR} EQUAL 0)
     message(FATAL_ERROR "Cannot determine tensorflow root, error code: ${TENSORFLOW_ROOT_RESULT_VAR}, error message: ${TENSORFLOW_ROOT_ERROR_VAR}")

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -156,17 +156,16 @@ if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
       NAMES ${module}
       PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
       )
-    message(STATUS ${TensorFlow_LIBRARY_${module}})
     if (TensorFlow_LIBRARY_${module})
       list(APPEND TensorFlow_LIBRARY ${TensorFlow_LIBRARY_${module}})
       get_filename_component(TensorFlow_LIBRARY_PATH_${module} ${TensorFlow_LIBRARY_${module}} PATH)
       list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlow_LIBRARY_PATH_${module}})
+      set (TensorFlow_LIBRARY_tensorflow_cc ${TensorFlow_LIBRARY_${module}})
     elseif (tensorflow_FIND_REQUIRED)
       message(FATAL_ERROR 
         "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' ")
     endif ()
   endforeach ()
-  set (TensorFlow_LIBRARY_tensorflow_cc ${TensorFlow_LIBRARY__pywrap_tensorflow_internal})
 endif()
 
 

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -42,7 +42,7 @@ if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS AND NOT SKBUILD)
     ERROR_VARIABLE TENSORFLOW_ROOT_ERROR_VAR
   )
   if (NOT ${TENSORFLOW_ROOT_RESULT_VAR} EQUAL 0)
-    message(FATAL_ERROR "Cannot determine tensorflow root, error message: ${TENSORFLOW_ROOT_ERROR_VAR}")
+    message(FATAL_ERROR "Cannot determine tensorflow root, error code: ${TENSORFLOW_ROOT_RESULT_VAR}, error message: ${TENSORFLOW_ROOT_ERROR_VAR}")
   endif()
 endif ()
 

--- a/source/cmake/Findtensorflow.cmake
+++ b/source/cmake/Findtensorflow.cmake
@@ -35,7 +35,7 @@ endif ()
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS AND NOT SKBUILD)
   # Here we try to install libtensorflow_cc.so as well as libtensorflow_framework.so using libs within the python site-package tensorflow folder.
   execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "import tensorflow; print(tensorflow.sysconfig.get_lib())"
+    COMMAND ${Python_EXECUTABLE} -c "import tensorflow; print(tensorflow.sysconfig.get_lib())"
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
     OUTPUT_VARIABLE TENSORFLOW_ROOT
     RESULT_VARIABLE TENSORFLOW_ROOT_RESULT_VAR
@@ -108,9 +108,9 @@ if (BUILD_CPP_IF AND NOT USE_TF_PYTHON_LIBS)
 	"You can manually set the tensorflow install path by -DTENSORFLOW_ROOT ")
     endif ()
   endforeach ()
-else (BUILD_CPP_IF)
+else (BUILD_CPP_IF AND NOT USE_TF_PYTHON_LIBS)
   message (STATUS "Disabled cpp interface build, looking for tensorflow_framework")
-endif (BUILD_CPP_IF)
+endif (BUILD_CPP_IF AND NOT USE_TF_PYTHON_LIBS)
 
 
 # tensorflow_framework
@@ -130,19 +130,19 @@ else ()
 list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .so.1)
 list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .so.2)
 endif()
-set (TensorFlow_LIBRARY_PATH "")
+set (TensorFlowFramework_LIBRARY_PATH "")
 foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
-  find_library(TensorFlow_LIBRARY_${module}
+  find_library(TensorFlowFramework_LIBRARY_${module}
     NAMES ${module}
     PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
     )
-  if (TensorFlow_LIBRARY_${module})
-    list(APPEND TensorFlow_LIBRARY ${TensorFlow_LIBRARY_${module}})
-    get_filename_component(TensorFlow_LIBRARY_PATH_${module} ${TensorFlow_LIBRARY_${module}} PATH)
-    list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlow_LIBRARY_PATH_${module}})
+  if (TensorFlowFramework_LIBRARY_${module})
+    list(APPEND TensorFlowFramework_LIBRARY ${TensorFlowFramework_LIBRARY_${module}})
+    get_filename_component(TensorFlowFramework_LIBRARY_PATH_${module} ${TensorFlowFramework_LIBRARY_${module}} PATH)
+    list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
   elseif (tensorflow_FIND_REQUIRED)
     message(FATAL_ERROR 
-      "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' "
+      "Not found ${TF_SUFFIX}${module} in '${TensorFlow_search_PATHS}' "
       "You can manually set the tensorflow install path by -DTENSORFLOW_ROOT ")
   endif ()
 endforeach ()
@@ -150,19 +150,19 @@ endforeach ()
 if (BUILD_CPP_IF AND USE_TF_PYTHON_LIBS)
   list(APPEND TensorFlow_search_PATHS ${TENSORFLOW_ROOT}/python)
   set(TF_SUFFIX "")
-  set(TensorFlowFramework_FIND_COMPONENTS _pywrap_tensorflow_internal)
-  foreach (module ${TensorFlowFramework_FIND_COMPONENTS})
-    find_library(TensorFlowFramework_LIBRARY_${module}
+  set(TensorFlow_FIND_COMPONENTS _pywrap_tensorflow_internal)
+  foreach (module ${TensorFlow_FIND_COMPONENTS})
+    find_library(TensorFlow_LIBRARY_${module}
       NAMES ${module}
       PATHS ${TensorFlow_search_PATHS} PATH_SUFFIXES ${TF_SUFFIX} NO_DEFAULT_PATH
       )
-    if (TensorFlowFramework_LIBRARY_${module})
-      list(APPEND TensorFlowFramework_LIBRARY ${TensorFlowFramework_LIBRARY_${module}})
-      get_filename_component(TensorFlowFramework_LIBRARY_PATH_${module} ${TensorFlowFramework_LIBRARY_${module}} PATH)
-      list(APPEND TensorFlowFramework_LIBRARY_PATH ${TensorFlowFramework_LIBRARY_PATH_${module}})
+    if (TensorFlow_LIBRARY_${module})
+      list(APPEND TensorFlow_LIBRARY ${TensorFlow_LIBRARY_${module}})
+      get_filename_component(TensorFlow_LIBRARY_PATH_${module} ${TensorFlow_LIBRARY_${module}} PATH)
+      list(APPEND TensorFlow_LIBRARY_PATH ${TensorFlow_LIBRARY_PATH_${module}})
     elseif (tensorflow_FIND_REQUIRED)
       message(FATAL_ERROR 
-        "Not found ${TF_SUFFIX}/${module} in '${TensorFlow_search_PATHS}' ")
+        "Not found ${TF_SUFFIX}${module} in '${TensorFlow_search_PATHS}' ")
     endif ()
   endforeach ()
 endif()

--- a/source/lmp/env_py.sh.in
+++ b/source/lmp/env_py.sh.in
@@ -5,9 +5,10 @@ PYTHON_LIBRARY_PATH="@Python_LIBRARY_DIRS@"
 
 TF_INCLUDE_DIRS=`echo $TENSORFLOW_INCLUDE_DIRS | sed "s/;/ -I/g"`
 TF_LIBRARY_PATH=`echo $TENSORFLOW_LIBRARY_PATH | sed "s/;/ -L/g"`
+PY_LIBRARY_PATH=`echo $PYTHON_LIBRARY_PATH | sed "s/;/ -L/g"`
 TF_RPATH=`echo $TENSORFLOW_LIBRARY_PATH | sed "s/;/ -Wl,-rpath=/g"`
 PYTHON_RPATH=`echo $PYTHON_LIBRARY_PATH | sed "s/;/ -Wl,-rpath=/g"`
 
 NNP_INC=" -D_GLIBCXX_USE_CXX11_ABI=@OP_CXX_ABI@ -std=c++@CMAKE_CXX_STANDARD@ -D@prec_def@ -DLAMMPS_VERSION_NUMBER=$(./lmp_version.sh) -I$TF_INCLUDE_DIRS -I$DEEPMD_ROOT/include/ "
-NNP_PATH=" -L$TF_LIBRARY_PATH -L$DEEPMD_ROOT/lib"
+NNP_PATH=" -L$TF_LIBRARY_PATH -L$DEEPMD_ROOT/lib -L$PY_LIBRARY_PATH"
 NNP_LIB=" -Wl,--no-as-needed -l@LIB_DEEPMD_CC@ -ltensorflow_cc -ltensorflow_framework -lpython@Python_VERSION_MAJOR@.@Python_VERSION_MINOR@ -Wl,-rpath=$TF_RPATH -Wl,-rpath=$DEEPMD_ROOT/lib -Wl,-rpath=$PYTHON_RPATH "

